### ISSUE TITLE
Exchange n messages and then exit

### ISSUE THREAD

**Issue by FedericoPonzi** [vi/websocat#65]
Is there any way to handle something like --one-message, but generalized to `n` messages? In my case, the webservice I'm contacting will respond back to my single message with two messages, and I don't want to be rude by closing the connection too early.

Thanks

**Comment by vi** [comment]
It can be implemented; not sure how useful would it be. Are there other cases when message limits can be beneficial?

> be rude by closing the connection too early.

Is it bad? Does it cause errors or warnings on server side? Or do you actually need content of the second message?

**Comment by edisinovcic** [comment]
This would be useful for my use case also. Having some kind of a termination clause (for example response containing a specific string in a response).

**Comment by FedericoPonzi** [comment]
> It can be implemented; not sure how useful would it be. Are there other cases when message limits can be beneficial?
> 
> > be rude by closing the connection too early.
> 
> Is it bad? Does it cause errors or warnings on server side? Or do you actually need content of the second message?

Yes, in my case that triggers an error on the server side. I've worked around it by using `| tail -n2`.

**Comment by vi** [comment]
Implemented relevant options and published a (not even a pre-release) build: https://github.com/vi/websocat/releases/tag/v2.0.0-alpha0

Please test.
